### PR TITLE
Add test configurations and README

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
$subject

## Approach
- Add README
- Add separate test configuration
- Fix public param issue

## Test environment
- SLP4
- Java 8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,75 @@
...
+
+   **Tip:** The following groups of test cases are available.<br>
+   Groups | Test Cases
+   ---| ---
+   connection | connection
+   pool | pool
+   transaction | transaction
+   execute | execute-basic <br> execute-params
+   batch-execute | batch-execute 
+   query | query-simple-params<br>query-numeric-params<br>query-complex-params
```
Comment (niveathika) [inline README.md:53]: Unnecssary

File: io-ballerina/build.gradle [hunk 2]
```diff
@@ -59,16 +59,69 @@ def tomlVersion = project.version.split("-")[0]
...
+    dependsOn(":io-native:test")
+    dependsOn(":io-test-utils:build")
+    dependsOn(unpackJballerinaTools)
+    dependsOn(updateTomlFile)
+    finalizedBy(revertTomlFile)
+
+    def groupParams = ""
+    if (project.hasProperty("groups")) {
+        groupParams = "--groups ${project.findProperty("groups")}"
+    }
```
Comment (niveathika) [inline io-ballerina/build.gradle:91]: Same

File: README.md (hunk 1 above)
Comment (BuddhiWathsala) [reply README.md]: Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/3/commits/5fe5749ebbad0844e3e310d0a23f0f004862e7b5

File: io-ballerina/build.gradle (hunk 2 above)
Comment (BuddhiWathsala) [reply io-ballerina/build.gradle]: Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/3/commits/5113b0bdf21654abb517506cfab4683d42820adb

**Review by niveathika — APPROVED** [review]

**Comment by niveathika** [comment]
@BuddhiWathsala Seems build is failing.

**Comment by BuddhiWathsala** [comment]
Fixed it and the windows build is failing due to a known issue

https://github.com/ballerina-platform/module-ballerina-io/issues/4

**Pull request merged**